File: prepper-bunker-frontend/src/app/models/medication.ts
export interface Medication {
  id: number;
  name: string;
  quantity: number;
  expirationDate: string;
  purpose: string;
  storageLocationId: number;
}

export interface MedicationCreate {
  name: string;
  quantity: number;
  expirationDate: string;
  purpose: string;
  storageLocationId: number;
}

File: prepper-bunker-backend/src/main/java/org/jonasfroeller/models/Medication.java
package org.jonasfroeller.models;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
@Table(name = "medication")
public class Medication extends PanacheEntity {
    public String name;
    public int quantity;

    @Column(name = "expiration_date")
    public LocalDate expirationDate;

    public String purpose;

    @ManyToOne
    @JoinColumn(name = "storage_location_id")
    public StorageLocation storageLocation;
}
